contrib/policy_upgrade/scan_mpf.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import json
+import hashlib
+
+from collections import OrderedDict
+
+
+def json_string(data):
+    return json.dumps(data, indent=2)
+
+
+def json_print(data):
+    print(json_string(data))
+
+
+def json_save(data, path):
+    string = json_string(data)
+    with open(path, "w") as f:
+        f.write(string)
```

Comment:
This is a lot of redundant function definitions.  All of these same functions are also defined in generate_hashes.py.  Violates SPOT (single point of truth).

Reply:
@mikeweilgart this is a WIP / draft. A code review is not that helpful at this time.

Reply:
Okay, that makes sense.  :)  Do you have any outline on the overall design you're planning to implement here?

(In the collaborative workflow I use, we write a design and then get a design review before we start implementation, so if there is anything like that I would be very interested to read the design doc.)

Reply:
@mikeweilgart There are some JIRA tickets already:

https://tracker.mender.io/browse/CFE-3121
https://tracker.mender.io/browse/CFE-3122
https://tracker.mender.io/browse/CFE-2991

However, they are mostly just placeholders, for now. We do have your comments from the previous PR:

https://github.com/cfengine/masterfiles/pull/1058

I will pick this up after 3.15 is released. I'll make sure to ping you whenever we have a prototype / design / proposal where it makes sense to ask for feedback.